roles/sap_general_preconfigure/README.md

```diff
@@ -1,341 +1,74 @@
-- _Type:_ `bool`
-- _Default:_ `true`
-
-Set to `false` if you do not want to modify the SELinux labels for the SAP directories set<br>
-in variable `sap_general_preconfigure_sap_directories`.<br>
-
-### sap_general_preconfigure_size_of_tmpfs_gb
-- _Type:_ `str`
-- _Default:_ `"{{ ((0.75 * (ansible_memtotal_mb + ansible_swaptotal_mb)) / 1024) | round | int }}"`
-
-The size of the tmpfs in GB. The formula used here is mentioned in SAP note 941735.<br>
-
-### sap_general_preconfigure_modify_etc_hosts
-- _Type:_ `bool`
-- _Default:_ `true`
-
-Set to `false` if you do not want the role to modify the `/etc/hosts` file.<br>
-
-### sap_general_preconfigure_etc_sysctl_sap_conf
-- _Type:_ `str`
-- _Default:_ `'/etc/sysctl.d/sap.conf'`
-
-The file name of the sysctl config file to be used<br>
-
-### sap_general_preconfigure_kernel_parameters
-- _Type:_ `list` with elements of type `dict`
-- _Default:_ (set by platform/environment specific variables)
-
-The Linux kernel parameters to use. By default, these are taken from the vars file.<br>
-The default for this parameter is set in the vars file which corresponds to the detected OS version.<br>
-
-Example:
-
-```yaml
-sap_general_preconfigure_kernel_parameters:
-- name: vm.max_map_count
-  value: '2147483647'
-- name: fs.aio-max-nr
-  value: '18446744073709551615'
-```
-
-### sap_general_preconfigure_max_hostname_length
-- _Type:_ `str`
-- _Default:_ `'13'`
-
-The maximum length of the hostname. See SAP note 611361.<br>
-
-### sap_hostname
-- _Type:_ `str`
-- _Default:_ `"{{ ansible_hostname }}"`
-
-The hostname to be used for updating or checking `/etc/hosts` entries.<br>
-
-### sap_domain
-- _Type:_ `str`
-- _Default:_ `"{{ ansible_domain }}"`
-
-The DNS domain name to be used for updating or checking `/etc/hosts` entries.<br>
-
-### sap_ip
-- _Type:_ `str`
-- _Default:_ `"{{ ansible_default_ipv4.address }}"`
-
-The IPV4 address to be used for updating or checking `/etc/hosts` entries.<br>
-
-### sap_general_preconfigure_db_group_name
-- _Type:_ `str`
-
-Use this variable to specify the name of the RHEL group which is used for the database processes.<br>
-If defined, it will be used to configure process limits as per step<br>
-Configuring Process Resource Limits<br>
-
-Example:
+This is general preconfigure role that used for both SAP Netweaver and SAP HANA, which have separate follow-up roles [sap_hana_preconfigure](https://github.com/sap-linuxlab/community.sap_install/tree/main/roles/sap_hana_preconfigure) and [sap_netweaver_preconfigure](https://github.com/sap-linuxlab/community.sap_install/tree/main/roles/sap_netweaver_preconfigure).
+<!-- END Description -->
 
+<!-- BEGIN Dependencies -->
+## Dependencies
+- `fedora.linux_system_roles`
+    - Roles:
+        - `selinux`
+- `community.sap_install` (This collection)
+    - Roles:
+        - `sap_maintain_etc_hosts`
+
+Install required collections by `ansible-galaxy install -vv -r meta/collection-requirements.yml`.
```

Comment:
I understand you mean here this file 

./roles/sap_general_preconfigure/meta/collection-requirements.yml

Aren't there are role requirements in this file, not collection requirements? File name is confusing isn't it?  

Why not use standard way to define requirements for the role, mentioned here

https://docs.ansible.com/ansible/latest/galaxy/user_guide.html#dependencies

Reply:
This is mainly meant for GitHub users reading Readme. Actual galaxy should list own dependencies you mentioned, but ...

@berndfinger We might need to add this because https://galaxy.ansible.com/ui/repo/published/community/sap_install/dependencies/ does not show dependency on linux_system_roles.

Reply:
@marcelmamula Agreed, it's advantageous to add the dependencies to the `dependencies` section in `galaxy.yml`. It should then also show up on the [Galaxy web page for this collection](https://galaxy.ansible.com/ui/repo/published/community/sap_install/dependencies/).

However, reading through related issues and discussions (https://github.com/sap-linuxlab/community.sap_install/issues/534, https://github.com/sap-linuxlab/community.sap_install/discussions/585, https://github.com/sap-linuxlab/community.sap_install/pull/552), there appear to be some disadvantages (however, maybe not really huge ones, or no longer existent).

@marcelmamula I'd like to get your opinion on adding the requirements in galaxy.yml.

Reply:
I agree with both points unfortunately.
1. Forcing dependency sounds simpler for us tracking it and having it visible for people who do not use GitHub
2. But linux-system-roles have dependency on ansible.posix, community.general, containers.podman . None of them have any dependencies.
3. There is also possibility that people using specific role, will not rely on LSR.

I am for option 1 to add it as galaxy dependency as those 3 are common and not dependent on others. @berndfinger